server/src/main/java/org/elasticsearch/action/admin/indices/settings/get/GetSettingsResponse.java

```diff
@@ -71,5 +118,121 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeString(cursor.key);
             Settings.writeSettingsToStream(cursor.value, out);
         }
+        if (out.getVersion().onOrAfter(org.elasticsearch.Version.V_7_0_0_alpha1)) {
+            out.writeVInt(indexToDefaultSettings.size());
+            for (ObjectObjectCursor<String, Settings> cursor : indexToDefaultSettings) {
+                out.writeString(cursor.key);
+                Settings.writeSettingsToStream(cursor.value, out);
+            }
+        }
+    }
+
+    private static void parseSettingsField(XContentParser parser, String currentIndexName, Map<String, Settings> indexToSettings,
+                                           Map<String, Settings> indexToDefaultSettings) throws IOException {
+        XContentParserUtils.ensureExpectedToken(XContentParser.Token.FIELD_NAME, parser.currentToken(), parser::getTokenLocation);
+        String settingsFieldName = parser.currentName(); //this should be either "settings" or "defaults"
+        parser.nextToken(); //go to settings object itself
+        XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser::getTokenLocation);
```

Comment:
This looks still too strict. Let's say we add something at this level that is not an object, parsing barfs. We should also test this scenario by letting the test insert random fields (it does so when supportsUnknownFields is set to true)